### ISSUE TITLE
Add support for Sourcehut CI

### ISSUE THREAD

**Issue by revelt** [watson/ci-info#56]
**Details:**
 - Ci name: Sourcehut
 - Website: https://builds.sr.ht/
 - Default env variables: nothing unique to Sourcehut
 - Test build with `is-ci` not detecting it's a CI: https://builds.sr.ht/~royston/job/359361

**Environment variables**
`JOB_ID`, `JOB_URL`

<details>
<summary>Test Build Manifest</summary>

```
image: alpine/edge
packages:
  - git
  - nodejs
  - npm
tasks:
  - setup: |
      npx is-ci && echo "This is a CI server" || echo "This is not a CI server"
      echo $JOB_ID
      echo $JOB_URL
      printenv
```

</details>

<details>
<summary>Output of Test Manifest</summary> 

```
npx: installed 2 in 1.752s
+ echo 'This is not a CI server'
This is not a CI server
+ echo 359361
359361
+ echo https://builds.sr.ht/~royston/job/359361
https://builds.sr.ht/~royston/job/359361
+ printenv
SHELL=/bin/sh
PWD=/home/build
LOGNAME=build
JOB_URL=https://builds.sr.ht/~royston/job/359361
HOME=/home/build
SSH_CONNECTION=172.17.0.1 55330 10.0.2.15 22
TERM=xterm-256color
USER=build
SHLVL=2
SSH_CLIENT=172.17.0.1 55330 22
PATH=/bin:/usr/bin:/sbin:/usr/sbin
JOB_ID=359361
MAIL=/var/mail/build
SSH_TTY=/dev/pts/0
_=/bin/printenv
```

</details>

**Additional context**
`env.JOB_URL.startsWith('https://builds.sr.ht/')` seems to be the pattern

**Comment by sibiraj-s** [comment]
Hi, need to know the default environment variables. something specific to Sourcehut. `JOB_ID` and `JOB_URL` are common vars used across CI and could result in false positives when used in other CI/CD's

**Comment by figgyc** [comment]
This seems to be where the environment is defined:
https://git.sr.ht/~sircmpwn/builds.sr.ht/tree/master/item/worker/tasks.go#L177
CI_NAME=sourcehut seems to be the most reliable way (it was only added [two months ago](https://git.sr.ht/~sircmpwn/builds.sr.ht/commit/81a5f43c01c8b84c713791c8630fc6acb16ac080#worker/tasks.go) which is why it doesn't appear in the example)
That seems to be all apart from JOB_URL and JOB_ID that the CI defines.

**Comment by sibiraj-s** [comment]
I don't see anything in their docs yet. See https://man.sr.ht/builds.sr.ht/#build-environment. I think it better to add only reliable variables. May be the docs is outdated? May be someone can raise an issue there to fix it.

**Comment by sibiraj-s** [comment]
Fixed in v3.7.0

**Comment by revelt** [comment]
good job!
